### DISCUSSION TITLE
Bevy usage in large webapp

### DISCUSSION THREAD

**Discussion by MNewbigging** [bevyengine/bevy#9134] [Q&A]
Hi folks, I'm doing a little research on Bevy as it might be a good fit for my workplace to use. I have a few questions revolving around our particular use case which I'm hoping someone here can shed some light on.

**Use Case**
We currently have a large Angular web app that wraps around a bespoke game engine which we use to render massive models and complex 3d scenes. The Angular app has a UI layer which reads from the state of the engine to display data about it, and also control the engine by means of manipulating the scene objects via an interface - as such the engine provides a large api for the ui layer to do this.

We're considering swapping our bespoke engine for Bevy, as it looks like it would provide a performance boost as well as abstract some of the lower level engine work.

**Questions**
Can we keep our Angular app and slot in a WASM build output from Bevy in a html canvas?

Can the Angular app read state from the Bevy build? 

Can the Angular app communicate (call functions from an api in) the Bevy build?

Could we implement a simple event listener in Bevy, for the Angular app to consume and respond to events?

Thanks for reading, appreciate any insights you might have :)

**Comment by MNewbigging** [comment]
Shameless bump!

**Comment by nicopap** [comment]
It's possible to get rust apps to communicate with web stuff through [`wasm-bindgen`](https://rustwasm.github.io/docs/wasm-bindgen/examples/hello-world.html) check out the link!

Concerning complexity: you would need to design a communication layer to get your web app to interact with bevy. Depending on the extent of what you need to do through the web interface, this might be a fairly tall order.

**Reply by MNewbigging** [reply]
Thanks for the link!
